drasyl-extras/src/main/java/org/drasyl/handler/connection/ConnectionConfig.java

```diff
@@ -0,0 +1,415 @@
+/*
+ * Copyright (c) 2020-2023 Heiko Bornholdt and Kevin Röbert
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.drasyl.handler.connection;
+
+import com.google.auto.value.AutoValue;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+
+import java.time.Duration;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.LongSupplier;
+
+import static java.time.Duration.ofMillis;
+import static java.time.Duration.ofSeconds;
+import static org.drasyl.handler.connection.TransmissionControlBlock.DRASYL_HDR_SIZE;
+
+@AutoValue
+public abstract class ConnectionConfig {
+    // Google Cloud applied MTU is 1460
+    static final int IP_MTU = 1460;
+    static final ConnectionConfig DEFAULT = new AutoValue_ConnectionConfig.Builder()
+            .issSupplier(Segment::randomSeq)
+            .sndBufSupplier(SendBuffer::new)
+            .rtnsQSupplier(channel -> new RetransmissionQueue())
+            .rcfBufSupplier(ReceiveBuffer::new)
+            .tcbSupplier((config, channel) -> new TransmissionControlBlock(
+                    config,
+                    0,
+                    0,
+                    0,
+                    0,
+                    0,
+                    0,
+                    config.sndBufSupplier().apply(channel),
+                    config.rtnsQSupplier().apply(channel),
+                    config.rcfBufSupplier().apply(channel),
+                    0,
+                    0,
+                    false
+            ))
+            .activeOpen(true)
+            .rmem(65_535 * 10)
+            // FIXME: change back to 2 minutes?
+            .msl(ofSeconds(2))
```

Comment:
?

Reply:
Siehe [Figure 5: TCP Connection State Diagram](https://www.rfc-editor.org/rfc/rfc9293.html#figure-5). Es geht um die Kante zwischen `TIME-WAIT` und `CLOSED`. Wir sollen hier 2*Maximum Segment Lifetime auf Nachrichten vom remote peer warten, bevor wir den `CLOSED` state betreten. Gemäß [3.4.2. Knowing When to Keep Quiet](https://www.rfc-editor.org/rfc/rfc9293.html#name-knowing-when-to-keep-quiet) beträgt die MSL 2 Minuten. Dies hat zur Folge, dass das Schließen des Netty Channels auf einer Seite immer 4 Minuten dauern würde. Daher habe ich den Wert verringert. Aber eigentlich finde ich die 4 Sekunden auch inakzeptabel. Ohne die Wartezeit kann es jedoch sein, dass einer der beiden schon eine neue Verbindung beginnen will, während der andere die vorherige noch schließt. Dann würde der Handshake fehlschlagen. Daher ja meine Überlegung. mit einem "virtuellen Port". Deine Meinung?

Reply:
Wenn das Hilft, warum nicht?